rust/simple/main.rs

```diff
@@ -32,20 +32,17 @@ fn main() {
 }
 
 fn try_main() -> Result<(), Box<dyn Error>> {
-    let stdin = io::stdin();
-    let stdin = stdin.lock();
     let mut counts: HashMap<String, u64> = HashMap::new();
-    for result in stdin.lines() {
-        let line = result?;
-        for word in line.split_whitespace() {
-            let canon = word.to_lowercase();
-            *counts.entry(canon).or_insert(0) += 1;
+    for line in io::stdin().lock().lines() {
+        for word in line?.split_whitespace() {
+            *counts.entry(word.to_lowercase()).or_insert(0) += 1;
         }
     }
 
-    let mut ordered: Vec<(String, u64)> = counts.into_iter().collect();
-    ordered.sort_by(|&(_, cnt1), &(_, cnt2)| cnt1.cmp(&cnt2).reverse());
-    for (word, count) in ordered {
+    let mut ordered: Vec<_> = counts.into_iter().collect();
+    ordered.sort_unstable_by_key(|&(_, count)| count);
```

Comment:
I kind of see `sort_untable_by_key` as an optimization, but it's debatable. I'll let @benhoyt decide.

As for sorting by key and printing in reverse, I do think that's a bit simpler and I don't think that's an optimization, so that seems fine to me I think.

Reply:
I originally think of the reason why use stable sort here is the tests require this. But seems unstable sort can also pass the tests so I did this. If we should not use unstable sort here, should it be ok to replace `sort_untable_by_key` with `sort_by_key`? I think it's more clear and maybe a little more efficient than `sort_by`.

Reply:
`sort_by_key` is fine.

Keeping the unstable sort in the other optimized programs sounds good though. Assuming you can see a measurable difference.

Reply:
Thanks for your review. In the current small test set there is indeed no significant performance improvement using unstable sort. IMO if we don't really need stable sort, it certainly has no negative effects on performance, nor does it increase the complexity of the code. So let's see @benhoyt 's decision. If stable sort actually is required I would replace it with `sort_by_key` then.

Reply:
I think it's okay with unstable, though you can argue it either way. 1) plain sort is shorter and since you don't care, simpler, or 2) it doesn't need to be stable, so in a sense unstable sort is a "simpler" fulfilling of the requirements. That said, I lean towards `sort_by_key` for simple.rs, so let's leave it at that.

Reply:
@benhoyt Updated in 3f81b8b. Also restore stable sort for the "bonus" solution.

Reply:
FWIW, I think an unstable sort in the bonus version is fair game. It isn't under any constraints really, and it has some minor optimizations in it already.

Reply:
> FWIW, I think an unstable sort in the bonus version is fair game. It isn't under any constraints really, and it has some minor optimizations in it already.

Make sense. Thanks for your quick reply. Updated in 7589c4a.